src/lualib/LuaRequire.ts

```diff
@@ -0,0 +1,14 @@
+const tstlpackage = { preload: {}, loaded: {} };
+
+function __TS__LuaRequire(this: void, moduleName: string): any {
+    if (!tstlpackage.loaded[moduleName]) {
+        const module: (this: void, module: string) => any = tstlpackage.preload[moduleName];
+        if (module) {
+            tstlpackage.loaded[moduleName] = module(moduleName);
```

Comment:
What about cyclic imports?

Reply:
Good opportunity to support that here. Those should work now